pyproject.toml

```
@@ -43,8 +43,9 @@ google-cloud-secret-manager = "^2.17.0"
 isodate = "~0.6.1"
 Jinja2 = "~3.1.2"
 jsonref = "~0.2"
-jsonschema = "~4.17.3"  # 4.18 has some significant breaking changes: https://github.com/python-jsonschema/jsonschema/releases/tag/v4.18.0
+jsonschema = ">=4.18.0,<5.0"
 packaging = "*"  # Transitive dependency used directly in code
+referencing = ">=0.30.0"  # Required for jsonschema 4.18+ RefResolver migration
```

aside - comment not necessary

```suggestion
referencing = ">=0.30.0"
```